### PR TITLE
implementing edit project button

### PR DESCRIPTION
**Pull request by LeenMass**

### CODE REVIEW

File: src/components/ContentsTable/index.jsx [hunk 1]
```diff
@@ -2,8 +2,18 @@ import React from 'react'
...
 import { BiArchiveIn } from 'react-icons/bi'
+import PopUpMessage from '../../pages/Projects/PopUpDialog'
+import { Dialog } from 'primereact/dialog'
+import { Button } from 'primereact/button'
 
 const ContentsTable = (props) => {
+  const [visible, setVisible] = React.useState(false)
+  const [Edit, setEdit] = React.useState({})
+
+  const EditRow = (Edit) => {
```
Comment (Karyum) [inline src/components/ContentsTable/index.jsx]: having `Edit` twice like this is quite confusing (first of all use camelCase, always camelCase unless it's a component or class name).
so maybe instead we can change to this ?
```suggestion
  const [edit, setEdit] = React.useState({})

  const EditRow = (rowData) => {
```

File: src/components/ContentsTable/index.jsx [hunk 2]
```diff
@@ -38,17 +48,35 @@ const ContentsTable = (props) => {
...
                 })}
               </div>
-              <button className={style.EditContainer}>
+              <Button
+                id="Edit"
+                className={style.EditContainer}
+                onClick={() => {
+                  if (EditRow) {
+                    EditRow(data)
+                    setVisible(true)
```
Comment (Karyum) [inline src/components/ContentsTable/index.jsx]: when someone clicks edit you are doing `setVisible(true)` twice, here and line 14

File: src/pages/Projects/PopUpDialog/index.jsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      const body = data
+
+      console.log(data)
+      await fetch(`${process.env.REACT_APP_API_URL}/AddingProject`, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(body)
+      })
+      if (body == null) {
+        return 'you must to insert data'
```
Comment (Karyum) [inline src/pages/Projects/PopUpDialog/index.jsx:34]: this return doesn't do anything really, for now maybe use [the toast from prime react](https://primereact.org/toast/#severity) as a fast way to display something for the user on error or on validation fail

File: src/pages/Projects/PopUpDialog/index.jsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  const createProject = async () => {
+    try {
+      const body = data
+
+      console.log(data)
+      await fetch(`${process.env.REACT_APP_API_URL}/AddingProject`, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(body)
+      })
```
Comment (Karyum) [inline src/pages/Projects/PopUpDialog/index.jsx]: the difference between fetch and Axios 😄, we do have it in the project.
You don't need to change it, I'm just showcasing Axios
```suggestion
      await axios.post(`${process.env.REACT_APP_API_URL}/AddingProject`, body)
```

File: src/pages/Projects/PopUpDialog/index.jsx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+          id="add"
+          label="add project"
+          icon="pi pi-check"
+          autoFocus
+          onClick={createProject}
+        />
+      </div>
+    )
+  } else if (props.clicked == 'Edit') {
+    return (
```
Comment (Karyum) [inline src/pages/Projects/PopUpDialog/index.jsx:105]: This way of doing 1 component for both edit and create is fine, but you could use the same form and same state for both the edit and create. this is not a critical change, it's just nice to have. 
We can arrange a short Zoom meeting and I can explain how to do it

File: src/pages/Projects/PopUpDialog/index.jsx [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+  const yyyy = date.getFullYear()
+  let mm = date.getMonth() + 1
+  let dd = date.getDate()
+  if (dd < 10) {
+    dd = '0' + dd
+  }
+  if (mm < 10) {
+    mm = '0' + mm
+  }
+  const formattedDate = dd + '/' + mm + '/' + yyyy
```
Comment (Karyum) [inline src/pages/Projects/PopUpDialog/index.jsx]: Checkout this package for date formatting https://www.npmjs.com/package/moment
it has a function called [`format()` ](https://momentjs.com/docs/#/displaying/format/) which can generate any date format

**Review by Karyum — CHANGES_REQUESTED** [review]
Good job on this PR 🎉

**Review by MuhammadAw90 — DISMISSED** [review]

**Review by saherhanif — DISMISSED** [review]

**Review by MuhammadAw90 — DISMISSED** [review]
The conflict fixed

File: src/pages/Projects/index.jsx [hunk 7]
```diff
@@ -1,39 +1,33 @@
...
-import api from '../../config'
 
 const Projects = () => {
   const [projects, setProjects] = useState([{}])
   const [visible, setVisible] = React.useState(false)
 
   const getProjects = async () => {
     try {
-      const result = await fetch(`${api.apiRequest}/ProjectPage`)
+      const result = await fetch('http://localhost:4000/Projects')
```
Comment (tawfiknasser) [inline src/pages/Projects/index.jsx]: localhost again

**Review by MuhammadAw90 — DISMISSED** [review]

File: src/pages/Projects/PopUpDialog/index.jsx (hunk 3 above)
Comment (tawfiknasser) [reply src/pages/Projects/PopUpDialog/index.jsx:34]: Agreed that they will add this in the upcoming PR.

File: src/pages/Projects/index.jsx [hunk 8]
```diff
@@ -4,14 +4,15 @@ import SearchBar from '../../components/SearchBar'
...
+import api from '../../config'
 
 const Projects = () => {
   const [projects, setProjects] = useState([{}])
   const [visible, setVisible] = React.useState(false)
 
   const getProjects = async () => {
     try {
-      const result = await fetch('http://localhost:4000/Projects')
+      const result = await fetch(`${api.apiRequest}/Projects`)
```
Comment (MuhammadAw90) [inline src/pages/Projects/index.jsx]: HERE replace to /ProjectPage

File: src/pages/Projects/index.jsx [hunk 9]
```diff
@@ -16,24 +12,23 @@ const Projects = () => {
 
   const getProjects = async () => {
     try {
-      const result = await fetch(`${api.apiRequest}/ProjectPage`)
+      const result = await fetch(`${api.apiRequest}/Projects`)
```
Comment (tawfiknasser) [inline src/pages/Projects/index.jsx]: project or projectPAge

File: src/pages/Projects/index.jsx [hunk 10]
```diff
@@ -46,7 +41,7 @@ const Projects = () => {
       <br />
       <div className={style.buttonsContainer}>
         <div className={style.Create}>
-          <Button label="Create Project" onClick={() => setVisible(true)} />
```
Comment (tawfiknasser) [inline src/pages/Projects/index.jsx:49]: why deleted ?

**Review by saherhanif — APPROVED** [review]

**Review by MuhammadAw90 — APPROVED** [review]

**Review by tawfiknasser — APPROVED** [review]

**Pull request merged**